libmamba/src/core/solution.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include "mamba/core/solution.hpp"
+
+namespace mamba
+{
+    Solution::Solution(const action_list& actions)
```

Comment:
Note that if you have only one constructor that takes by value and move the copy in, it's equivalent to the 3 constructors here, as this code cannot be inlined.

Reply:
Potentially one extra move, no?